building-epics/getting-epics7-on-rtems5-beatnik.rst

```diff
@@ -0,0 +1,4365 @@
+   ok  3 - epicsRingPointerGetFree(ring)==rsize
+   ok  4 - epicsRingPointerGetSize(ring)==rsize
+   ok  5 - epicsRingPointerGetUsed(ring)==0
+   ok  6 - epicsRingPointerGetHighWaterMark(ring)==0
+   ok  7 - epicsRingPointerPop(ring)==NULL
+   ok  8 - epicsRingPointerPush(ring, addr)==1
+   ok  9 - !epicsRingPointerIsEmpty(ring)
+   ok 10 - !epicsRingPointerIsFull(ring)
+   ok 11 - epicsRingPointerGetFree(ring)==rsize-1
+   ok 12 - epicsRingPointerGetSize(ring)==rsize
+   ok 13 - epicsRingPointerGetUsed(ring)==1
+   ok 14 - epicsRingPointerGetHighWaterMark(ring)==1
+   ok 15 - epicsRingPointerGetHighWaterMark(ring)==1
+   # Fill it up
+   ok 16 - 101 == 101
+   ok 17 - !epicsRingPointerIsEmpty(ring)
+   ok 18 - epicsRingPointerIsFull(ring)
+   ok 19 - epicsRingPointerGetFree(ring)==0
+   ok 20 - epicsRingPointerGetSize(ring)==rsize
+   ok 21 - epicsRingPointerGetUsed(ring)==rsize
+   ok 22 - epicsRingPointerGetHighWaterMark(ring)==rsize
+   # Drain it out
+   ok 23 - !foundCorruption
+   ok 24 - 101 == 101
+   ok 25 - epicsRingPointerIsEmpty(ring)
+   ok 26 - !epicsRingPointerIsFull(ring)
+   ok 27 - epicsRingPointerGetFree(ring)==rsize
+   ok 28 - epicsRingPointerGetSize(ring)==rsize
+   ok 29 - epicsRingPointerGetUsed(ring)==0
+   ok 30 - epicsRingPointerGetHighWaterMark(ring)==rsize
+   # Fill it up again
+   # flush
+   ok 31 - epicsRingPointerIsFull(ring)
+   ok 32 - epicsRingPointerIsEmpty(ring)
+   ok 33 - !epicsRingPointerIsFull(ring)
+   ok 34 - epicsRingPointerGetFree(ring)==rsize
+   ok 35 - epicsRingPointerGetSize(ring)==rsize
+   ok 36 - epicsRingPointerGetUsed(ring)==0
+   # single producer, single consumer without locking
+   # Everything enqueued, Stopping consumer
+   # Pushed 1000, have 0 remaining unconsumed
+   # Expect 1000 consumed
+   ok 37 - 0x3e703e7 == 0x3e703e7
+   ok 38 - Consumer consumed all
+   ok 39 - !foundCorruption
+   # single producer, single consumer with locking
+   # Everything enqueued, Stopping consumer
+   # Pushed 1000, have 0 remaining unconsumed
+   # Expect 1000 consumed
+   ok 40 - 0x3e703e7 == 0x3e703e7
+   ok 41 - Consumer consumed all
+   ok 42 - !foundCorruption
+
+       Results
+       =======
+          Tests: 42
+         Passed:  42 = 100.00%
+
+   ***** taskwdTest *****
+   1..8
+   ok  1 - monInsert(thread='testTask1')
+   ok  2 - monInsert(thread='testTask2')
+   # Task 2 suspending
+   ok  3 - monNotify(thread='testTask2', suspended=1)
+   ok  4 - anyNotify(thread='testTask2')
+   ok  5 - taskNotify id='2'
+   # Task 2 alive again
+   ok  6 - monNotify(thread='testTask2', suspended=0)
+   # Task 2 cleaning up
+   ok  7 - monRemove(thread='testTask2')
+   # Task 1 cleaning up
+   ok  8 - monRemove(thread='testTask1')
+
+       Results
+       =======
+          Tests: 8
+         Passed:   8 = 100.00%
+
+
+       EPICS Test Harness Results
+       ==========================
+
+   All tests successful.
+   Programs=37, Tests=3062, 352 wallclock secs
```

Comment:
Note that the section just above here ("EPICS Test Harness Results") is showing that all the above libCom test programs passed. Unfortunately in the libCom tests we have to run epicsExitTest() afterwards so it isn't obvious that there _is_ a final summary of the results. I agree with Ralph that you could cut out all of the results above and just show this bit (this document is a bit intimidating due to its length). Maybe place the full results on a separate page, and similarly have pages with the results from all the other test harnesses in the other test areas (this was just testing libCom)?